api/api/src/main/scala/uk/ac/wellcome/platform/api/services/CollectionRequestBuilder.scala

```diff
@@ -0,0 +1,79 @@
+package uk.ac.wellcome.platform.api.services
+
+import com.sksamuel.elastic4s.ElasticDsl._
+import com.sksamuel.elastic4s.Index
+import com.sksamuel.elastic4s.requests.searches.queries.Query
+import com.sksamuel.elastic4s.requests.searches.SearchRequest
+
+/**
+  * Builds an elasticsearch request for fetching nodes in some Collection tree.
+  */
+case class CollectionRequestBuilder(index: Index,
+                                    expandedPaths: List[String],
+                                    maxNodes: Int = 1000) {
+
+  val excludeFields = List(
+    "data.items",
+    "data.notes",
+    "data.production",
+    "data.subjects",
+    "data.genres",
+    "data.images",
+  )
+
+  lazy val request: SearchRequest =
+    search(index)
+      .query(query)
+      .from(0)
+      .limit(maxNodes)
+      .sourceExclude(excludeFields)
+
+  lazy val query: Query =
+    should(expandedPaths.map(expandPathQuery))
+
+  /**
+    * Get the node at some path, it's direct children, it's ancestors, and the
```

Comment:
nit: no apostrophe in possessive "its"

```suggestion
    * Get the node at some path, its direct children, its ancestors, and the
```